src/hooks/useCensus3DaoMembers.tsx

```diff
@@ -34,13 +36,16 @@ export const useCensus3DaoMembers = ({
   const {id: proposalId} = useParams();
   const {data: daoToken} = useDaoToken(pluginAddress);
   const {address} = useWallet();
+  const [tokenSynced, setTokenSynced] = useState(false);
 
   // If is not a wrapped token and not on a proposal context we can still get the token holders amount
-  const enableCensus3Token = enable && !proposalId;
+  const enableCensus3Token = enable && !!daoToken?.address && !proposalId;
   const {data: census3Token} = useCensus3Token(
     {tokenAddress: daoToken?.address ?? ''},
     {
-      enabled: enable && !!daoToken?.address && enableCensus3Token,
+      enabled: enableCensus3Token,
+      refetchInterval:
+        enableCensus3Token && !tokenSynced ? REFETCH_INTERVAL_MS : false,
```

Comment:
`refetchInterval` takes a function which it passes the latest data to. Using this, we can get rid of the useEffect + useState tracking whether the token is synced

```suggestion
        refetchInterval: token =>
        token?.status.synced ? false : REFETCH_INTERVAL_MS,
```

Reply:
5215dc81